src/ol/imagebase.js

```diff
@@ -53,7 +53,7 @@ ol.ImageBase = function(extent, resolution, pixelRatio, state, attributions) {
 
   /**
    * @protected
-   * @type {number|undefined}
+   * @type {Array.<number>|number|undefined}
```

Comment:
Looks like you could remove a lot of duplicated `if (goog.isArray(imageResolution))` code if you'd always have an `Array.<number>` here.